rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.create/20_synthetic_source.yml

```diff
@@ -1214,3 +1224,231 @@ fallback synthetic_source for text field:
+  - requires:
+      cluster_features: ["mapper.source.synthetic_source_with_copy_to_and_doc_values_false"]
+      reason: requires copy_to support in synthetic source
+
+  - do:
+      indices.create:
+        index: test
+        body:
+          mappings:
+            _source:
+              mode: synthetic
+            properties:
+              name:
+                type: keyword
+              k:
+                type: keyword
+                ignore_above: 1
+                copy_to: copy
+              long:
+                type: long
+                ignore_malformed: true
+                copy_to: copy
+              copy:
+                type: keyword
+
+  - do:
+      index:
+        index: test
+        id: 1
+        refresh: true
+        body:
+          name: "A"
+          k: "hello"
+          long: "world"
+
+  - do:
+      index:
+        index: test
+        id: 2
+        refresh: true
+        body:
+          name: "B"
+          k: ["5", "6"]
+          long: ["7", "8"]
+
+  - do:
+      search:
+        index: test
+        sort: name
+        body:
+          docvalue_fields: [ "copy" ]
+
+  - match:
+      hits.hits.0._source:
+        name: "A"
+        k: "hello"
+        long: "world"
+  - match: { hits.hits.0.fields.copy: ["hello", "world"] }
+
+  - match:
+      hits.hits.1._source:
+        name: "B"
+        k: ["5", "6"]
+        long: ["7", "8"]
+  - match: { hits.hits.1.fields.copy: ["5", "6", "7", "8"] }
+
+
+---
+synthetic_source with copy_to field having values in source:
+  - requires:
+      cluster_features: ["mapper.source.synthetic_source_with_copy_to_and_doc_values_false"]
+      reason: requires copy_to support in synthetic source
+
+  - do:
+      indices.create:
+        index: test
+        body:
+          mappings:
+            _source:
+              mode: synthetic
+            properties:
+              name:
+                type: keyword
+              k:
+                type: keyword
+                copy_to: copy
+              copy:
+                type: keyword
+
+  - do:
+      index:
+        index: test
+        id: 1
+        refresh: true
+        body:
+          name: "A"
+          copy: "world"
+          k: "hello"
+
+  - do:
+      index:
+        index: test
+        id: 2
+        refresh: true
+        body:
+          name: "B"
+          k: ["5", "6"]
+          copy: ["7", "8"]
+
+  - do:
+      search:
+        index: test
+        sort: name
+        body:
+          docvalue_fields: [ "copy" ]
+
+  - match:
+      hits.hits.0._source:
+        name: "A"
+        k: "hello"
+        copy: "world"
+  - match: { hits.hits.0.fields.copy: ["hello", "world"] }
+
+  - match:
+      hits.hits.1._source:
+        name: "B"
+        k: ["5", "6"]
+        copy: ["7", "8"]
+  - match: { hits.hits.1.fields.copy: ["5", "6", "7", "8"] }
+
+---
+synthetic_source with copy_to field from dynamic template having values in source:
```

Comment:
Can we check that using `copy_to` results in error when used with object-like fields? Note: it is not just `object` and `nested` but also things `range` or geo-like. I think this is important for BWC...we need to keep the error-behaviour.